### PR TITLE
treestartr blog post

### PR DESCRIPTION
**Pull request by wrightaprilm**
Hi all (and ping @stefaniebutland),

Here's a draft of my treestartr post. I might make a few more tweaks (like getting new lab members on to my site and actually linking to their bios in the piece).  Let me know if this is roughly what you're looking for.

### CODE REVIEW

**Comment by stefaniebutland** [comment]
Thank you for submitting this @wrightaprilm. Quick scan gave me a very big smile. I'll do a proper review by end of day Tues Dec 4.

File: content/blog/2018-12-11-treestartr.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+In these cases, we know roughly where on the tree might be a reasonable starting place to try attaching the taxon. In this case, we can use the function [`text_placr`](https://ropensci.github.io/treeStartR/#adding-tips-via-csv) to read in a dataframe of taxonomic relationships. We can then use those relationships to generate a starting tree. An example taxonomy dataframe is below. In this data frame the two tips to be added, _Kretzoiartcos beatrix_ and _Ursus abstrusus_, are listed in the `taxon` column. The `clade` column contains tips that are currently on the tree. The two tips associated with _Kretzoiartcos beatrix_, _Indarctos arctoides_ and _Indarctos vireti_, will form a monophyletic group with _Kretzoiartcos beatrix_ on the generated starting tree. The three taxa associated with _Ursus abstrusus_ will form a monophyletic group with _Ursus abstrusus_. 
+
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: Any chance you could add a figure/ sketch of the tree you are describing in the text so that readers can picture "tree", "tips", "monophyletic group"? I'm familiar with the terms but still had trouble picturing it from the data frame and text without getting distracted from the flow of the post.

File: content/blog/2018-12-11-treestartr.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+### Adding tips by hand
+
+For the visual among us, [`absent_tippr`](https://ropensci.github.io/treeStartR/#adding-tips-at-random) opens a graphical user interface. Users can then see the computer-generated node label for every node in the tree, and choose the one they would like the added tip to subtend.
+
+
+### Mix and Match
+
+For some of our workflows in the lab, we use multiple functions. For example, we might first place all the tips that have close relatives using `present_tippr`. Then, tips that represent more distantly related taxa can be placed via `text_placr`, and the leftover fossils added via `rand_absent_tippr`. 
+
+## Why ROpenSci
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: ROpenSci -> rOpenSci 
throughout

File: content/blog/2018-12-11-treestartr.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
...itly R (so the editor and reviewers would know best practices) and the guidelines and how review would proceed were very clear. And for a totally new experience of in-depth code review, being able to see where the effort was collaborative was helpful. I looked at [other journals](https://joss.theoj.org/about#author_guidelines), but I liked that ROpenSci had biologists as editors. Because the handling editor ([Scott Chamberlain](https://scottchamberlain.info/)) is a biologist, he had the ability to work with me to find reviewers ([David Bapst](https://geogeo.tamu.edu/people/faculty/bapstdavidwilliam) and [Rachel Warnock](https://www.bsse.ethz.ch/cevo/the-group/people/person-detail.html?persid=215787)) who are serious experts in both R and the particular type of phylogenetics that I do. I chose Dave & Rachel because I know I can collaborate with them, and they will be constructive and kind. But they will also really try to break the package and dig in to the meat of what software functions are required to support the science we do, and tell me honestly if my package is providing those functions. [You can see](https://github.com/ropensci/onboarding/issues/239) they weren't easy on me!
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: Note the content at the JOSS link (https://joss.theoj.org/about#author_guidelines) has a new location https://joss.readthedocs.io/en/latest/submitting.html

File: content/blog/2018-12-11-treestartr.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+ROpenSci also has a collaboration with [Methods in Ecology and Evolution](https://besjournals.onlinelibrary.wiley.com/hub/journal/2041210X/author-guidelines). I plan to send a short software note there. It's very nice to have the flexibility to send the paper to a traditional outlet like MEE, or to send it to [JOSS](https://joss.theoj.org/), or to do nothing at all, if I didn't want to.
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: maybe link `[collaboration](https://ropensci.org/blog/2017/11/29/review-collaboration-mee/)`

please spell MEE the second time, rather than abbreviation

File: content/blog/2018-12-11-treestartr.md (hunk 3 above)
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: " I chose Dave & Rachel ..." Could you say "suggested"?

File: content/blog/2018-12-11-treestartr.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+### tl;dr
+
+Come on in, biologists. The water's fine. 
+
+## Future
+
+I'll likely add in more utility functions for parsing biological data for use with divergence time estimation. Particularly, I often work with [RevBayes](https://revbayes.github.io/), which is a sandbox-like framework for Bayesian analysis of phylogeny and comparative methods. Our analyses integrate across fossil taxa from databases, molecular data from the literature and stratigraphic ranges. Each data source has special and unique quirks, and I'm always looking for ways to lower the bar for undergraduate researchers to be able to access these sorts of complex modeling frameworks. I have a [Contributing](https://github.com/ropensci/treeStartR/blob/master/.github/CONTRIBUTING.md) guide here; please feel free to make a pull request or file an issue if there's something you'd like to see added.
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]: Can you change the order of these phrases? It sounds like you want pull requests to your contributing guide
" I have a Contributing guide here; please feel free to make a pull request or file an issue if there’s something you’d like to see added."

File: content/blog/2018-12-11-treestartr.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+slug: "treestartr"
+title: Generating reasonable starting trees for complex phylogenetic analyses
+package_version: 0.1.0
+authors:
+  - name: April Wright
+    url: https://ropensci.github.io/treeStartR/
+date: 2018-12-11
+categories: blog
+topicid: TODO
+tags:
```
[the remark this answers is not in the document]
Comment (wrightaprilm) [reply content/blog/2018-12-11-treestartr.md]:
```suggestion
tags:
- software-peer-review
```

File: content/blog/2018-12-11-treestartr.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
+---
+slug: "treestartr"
+title: Generating reasonable starting trees for complex phylogenetic analyses
+package_version: 0.1.0
+authors:
+  - name: April Wright
+    url: https://ropensci.github.io/treeStartR/
```
Comment (wrightaprilm) [inline content/blog/2018-12-11-treestartr.md]:
```suggestion
    url: https://paleantology.com/the-wright-lab/
```

File: content/blog/2018-12-11-treestartr.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
+---
+slug: "treestartr"
+title: Generating reasonable starting trees for complex phylogenetic analyses
+package_version: 0.1.0
+authors:
+  - name: April Wright
+    url: https://paleantology.com/the-wright-lab/
```
Comment (wrightaprilm) [inline content/blog/2018-12-11-treestartr.md]:
```suggestion
    url: https://paleantology.com/the-wright-lab/
    twitter: WrightingApril
```

File: content/blog/2018-12-11-treestartr.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (wrightaprilm) [reply content/blog/2018-12-11-treestartr.md]: Ooooh, good catch.

**Comment by stefaniebutland** [comment]
@wrightaprilm Please let me know when you're finished addressing comments - any time up to Mon Dec 10 - and I'll have a last look before publishing.

**Comment by wrightaprilm** [comment]
I think I've addressed all your comments. Go ahead and look it over - I also sent it to the reviewers to have a look, since they're mentioned quite a bit towards the end.

**Comment by wrightaprilm** [comment]
OK, heard back from all the package reviewers and made changes as suggested by them. I'm happy if you're happy!

**Comment by stefaniebutland** [comment]
> heard back from all the package reviewers and made changes as suggested by them

This is great!!

I think a few of my suggestions are in unresolved "hidden conversations". I'm sorry about that - it's possible that my way of adding comments isn't clear enough to the person receiving the review. Copying them here.

screen shot 2018-12-07 at 2 55 35 pm

- "Ants have tons of fossils" Can you rephrase to indicate there are lots of fossilized ants?
I don't want to alter your writing style so you can choose to ignore this.
- Can you add a line to say what you do once you have a starter tree? I (somewhat naive reader) felt like that was missing in the emphasis on getting a starter tree
- "But not everyone else does." Can you omit this line? It's a fact, but in reading it, I felt like something else related to it was coming up in the text
- Any chance you could add a figure/ sketch of the tree you are describing in the text so that readers can picture "tree", "tips", "monophyletic group"? I'm familiar with the terms but still had trouble picturing it from the data frame and text without getting distracted from the flow of the post.

This last request might be a bit much to request at this late stage (Friday!) if you don't already have something ready, so it's a wishlist item for clarity but I won't insist on it! 

It's a really nicely written post so it's 100% fine to say "let's publish without further changes".

**Comment by wrightaprilm** [comment]
Oh, no problem, I actually have a great figure for there being tons of ants. Always excited to use it. 

I'm a little old school with github, and sometimes miss multiple comments. Sometimes I feel a little bad for my Data Carpentry dc-ecology-py comaintainer ;P

**Comment by stefaniebutland** [comment]
> have a great figure for there being tons of ants.

key issue is for non-phylo familiar folk to see what the terms mean and therefore be able to picture where the placements in your table would go

**Comment by wrightaprilm** [comment]
I ended up not adding the ant figure, but contextualizing what it means to have lots of fossils, and why that's important for my personal research. Also added a phylogeny figure.

File: content/blog/2018-12-11-treestartr.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+![](../../themes/ropensci/static/img/blog-images/2018-121-11-treestartr/tree.png)
```
Comment (stefaniebutland) [inline content/blog/2018-12-11-treestartr.md]:
```suggestion
![](/img/blog-images/2018-12-11-treestartr/tree.png)
```

**Comment by stefaniebutland** [comment]
@wrightaprilm the added ant fossils text is great.

I made a "suggested change" in your pull request to fix typo in path to phylo fig (date part of folder name) that prevented it from being viewed, and also changed the way the path is listed. The "suggested change" is a new GitHub feature that, if you accept, is automatically committed (only the 2nd time I've used this). fyi suggested path is ``.

You can check the final product in preview: https://deploy-preview-353--ropensci.netlify.com/blog/2018/12/11/treestartr/

If for some reason the image doesn't look right (size etc), I can help, and maybe can make the changes myself to save you the back and forth.

Cheers!

**Comment by wrightaprilm** [comment]
I love the suggestion mechanism!

It would have been such a clean fix ... if the actual name of the directory didn't contain a typo. Committed a correctly-named directory and removed my typo'd one. Render looks nice to me.
